Migration errors with new sqlx version

@klochowicz : did you run the migration when merging this PR? Looks like we get errors when running migration since this PR: 

```
╰ ./prepare_db.sh
Applied 20210903050345/migrate create cfd and order tables (62.936154ms)
Applied 20211025050345/migrate rename term to settlement time interval (1.782077ms)
Applied 20211207035936/migrate add-fee-to-order (651.475µs)
Applied 20211208000000/migrate associate counterpart identity with cfd (5.464037ms)
Applied 20211210000000/migrate remove-order-from-database (1.242287ms)
Applied 20211213000000/migrate remove-oracle event id from cfd (732.646µs)
Applied 20211214055426/migrate replace-origin-with-role (621.597µs)
Applied 20211220000000/migrate introduce-event-sourcing (1.013698ms)
Applied 20220121052945/migrate add opening fee (498.827µs)
Applied 20220124063030/migrate add initial funding rate (706.875µs)
Applied 20220125051124/migrate add initial tx fee rate (591.321µs)
Applied 20220404113202/migrate create closed cfds tables (32.344378ms)
Applied 20220428103201/migrate create failed cfds table (32.142125ms)
Applied 20220502172229/migrate create time to first position table (33.394799ms)
Applied 20220516000538/migrate counterparty peer id (42.749273ms)
Applied 20220525225235/migrate rename tables (45.511744ms)
Applied 20220526225235/migrate rollover-completed-table (32.92339ms)
Applied 20220622041906/migrate add settlement event id to revoke commit transactions (12.677079ms)
Applied 20220622055305/migrate add complete fee to rollover tables (14.398834ms)
   Compiling model v0.1.0 (/Users/bonomat/src/github/comit/hermes/model)
   Compiling sqlite-db v0.1.0 (/Users/bonomat/src/github/comit/hermes/sqlite-db)
error[E0308]: mismatched types
   --> sqlite-db/src/closed.rs:854:15
    |
854 |       let row = sqlx::query_as!(
    |  _______________^
855 | |         models::Settlement::Cet,
856 | |         r#"
857 | |         SELECT
...   |
872 | |         id
873 | |     )
    | |_____^ expected struct `models::Txid`, found enum `std::option::Option`
    |
    = note: expected struct `models::Txid`
                 found enum `std::option::Option<models::Txid>`
    = note: this error originates in the macro `$crate::sqlx_macros::expand_query` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
   --> sqlite-db/src/closed.rs:886:15
    |
886 |       let row = sqlx::query_as!(
    |  _______________^
887 | |         models::Settlement::Refund,
888 | |         r#"
889 | |         SELECT
...   |
903 | |         id
904 | |     )
    | |_____^ expected struct `models::Txid`, found enum `std::option::Option`
    |
    = note: expected struct `models::Txid`
                 found enum `std::option::Option<models::Txid>`
    = note: this error originates in the macro `$crate::sqlx_macros::expand_query` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
   --> sqlite-db/src/closed.rs:976:23
    |
976 |     Ok(Timestamp::new(row.created_at))
    |                       ^^^^^^^^^^^^^^ expected `i64`, found enum `std::option::Option`
    |
    = note: expected type `i64`
               found enum `std::option::Option<i64>`

error[E0308]: mismatched types
   --> sqlite-db/src/failed.rs:314:23
    |
314 |     Ok(Timestamp::new(row.created_at))
    |                       ^^^^^^^^^^^^^^ expected `i64`, found enum `std::option::Option`
    |
    = note: expected type `i64`
               found enum `std::option::Option<i64>`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `sqlite-db` due to 4 previous errors
error: `cargo check` failed with status: exit status: 101
```

_Originally posted by @bonomat in https://github.com/itchysats/itchysats/issues/2256#issuecomment-1171954182_

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -26,3 +26,6 @@ xtra = { git = "https://github.com/comit-network/xtra" } # We need to use unrele
 maia = { git = "https://github.com/comit-network/maia", rev = "fc6b78b98407b10b55f8cfd152062ad77f98cd9f" } # Unreleased
 maia-core = { git = "https://github.com/comit-network/maia", tag = "0.1.1", package = "maia-core" } # Pinned to support maia 0.1 and 0.2
 xtra_productivity = { git = "https://github.com/comit-network/xtra-productivity" } # Unreleased
+
+[profile.dev.package.sqlx-macros]
+opt-level = 3

--- a/sqlite-db/sqlx-data.json
+++ b/sqlite-db/sqlx-data.json
@@ -70,24 +70,6 @@
       ]
     }
   },
-  "12a758d80dca77c52d5a17c9c2b286bdc537dee587289bf6eba94bf0fb16fc83": {
-    "query": "\n        SELECT\n            event_log.created_at\n        FROM\n            event_log\n        JOIN\n            closed_cfds on closed_cfds.id = event_log.cfd_id\n        WHERE\n            closed_cfds.uuid = $1\n        ORDER BY event_log.created_at ASC\n        LIMIT 1\n        ",
-    "describe": {
-      "columns": [
-        {
-          "name": "created_at",
-          "ordinal": 0,
-          "type_info": "Int64"
-        }
-      ],
-      "parameters": {
-        "Right": 1
-      },
-      "nullable": [
-        false
-      ]
-    }
-  },
   "1b0a898c7975e9d9eb313fe900b15c4672f11ff110e0ae16560d7f3d7b4fddc4": {
     "query": "\n            select\n                id as cfd_id,\n                uuid as \"uuid: models::OrderId\",\n                position as \"position: models::Position\",\n                initial_price as \"initial_price: models::Price\",\n                leverage as \"leverage: models::Leverage\",\n                settlement_time_interval_hours,\n                quantity_usd as \"quantity_usd: models::Usd\",\n                counterparty_network_identity as \"counterparty_network_identity: models::Identity\",\n                counterparty_peer_id as \"counterparty_peer_id: models::PeerId\",\n                role as \"role: models::Role\",\n                opening_fee as \"opening_fee: models::OpeningFee\",\n                initial_funding_rate as \"initial_funding_rate: models::FundingRate\",\n                initial_tx_fee_rate as \"initial_tx_fee_rate: models::TxFeeRate\"\n            from\n                cfds\n            where\n                cfds.uuid = $1\n            ",
     "describe": {
@@ -244,6 +226,24 @@
       ]
     }
   },
+  "3b946031bc9598649255793e28a2c34538daec61b075f890757967eef04ff1af": {
+    "query": "\n        SELECT\n            event_log.created_at as \"created_at!: i64\"\n        FROM\n            event_log\n        JOIN\n            closed_cfds on closed_cfds.id = event_log.cfd_id\n        WHERE\n            closed_cfds.uuid = $1\n        ORDER BY event_log.created_at ASC\n        LIMIT 1\n        ",
+    "describe": {
+      "columns": [
+        {
+          "name": "created_at!: i64",
+          "ordinal": 0,
+          "type_info": "Int64"
+        }
+      ],
+      "parameters": {
+        "Right": 1
+      },
+      "nullable": [
+        true
+      ]
+    }
+  },
   "450bfcea8dcac5288b69187eb4ae5aec72012d7320e1d4d2602c448671512295": {
     "query": "\n            delete from rollover_completed_event_data where cfd_id = (select id from cfds where cfds.uuid = $1)\n        ",
     "describe": {
@@ -300,6 +300,24 @@
         "Right": 1
       },
       "nullable": []
+    }
+  },
+  "697d9ca427dd0d3d8b3a21640bb2f309d30fb34b8d57bf663fe282892b21dd5d": {
+    "query": "\n        SELECT\n            event_log_failed.created_at as \"created_at!: i64\"\n        FROM\n            event_log_failed\n        JOIN\n            failed_cfds on failed_cfds.id = event_log_failed.cfd_id\n        WHERE\n            failed_cfds.uuid = $1\n        ORDER BY event_log_failed.created_at ASC\n        LIMIT 1\n        ",
+    "describe": {
+      "columns": [
+        {
+          "name": "created_at!: i64",
+          "ordinal": 0,
+          "type_info": "Int64"
+        }
+      ],
+      "parameters": {
+        "Right": 1
+      },
+      "nullable": [
+        true
+      ]
     }
   },
   "6a0e947eb3254ed994e44b0018ca9232d37bc405b37856b15be5a0ef81757af1": {
@@ -410,6 +428,48 @@
         "Right": 3
       },
       "nullable": []
+    }
+  },
+  "8376057fa3aeb05be23c79c288b929b9b6a8590b410f2496ab2c52dba24b5e8c": {
+    "query": "\n        SELECT\n            closed_commit_txs.txid as \"commit_txid!: models::Txid\",\n            closed_cets.txid as \"txid: models::Txid\",\n            closed_cets.vout as \"vout: models::Vout\",\n            closed_cets.payout as \"payout: models::Payout\",\n            closed_cets.price as \"price: models::Price\"\n        FROM\n            closed_cets\n        JOIN\n            closed_commit_txs on closed_commit_txs.cfd_id = closed_cets.cfd_id\n        JOIN\n            closed_cfds on closed_cfds.id = closed_cets.cfd_id\n        WHERE\n            closed_cfds.uuid = $1\n        ",
+    "describe": {
+      "columns": [
+        {
+          "name": "commit_txid!: models::Txid",
+          "ordinal": 0,
+          "type_info": "Text"
+        },
+        {
+          "name": "txid: models::Txid",
+          "ordinal": 1,
+          "type_info": "Text"
+        },
+        {
+          "name": "vout: models::Vout",
+          "ordinal": 2,
+          "type_info": "Int64"
+        },
+        {
+          "name": "payout: models::Payout",
+          "ordinal": 3,
+          "type_info": "Int64"
+        },
+        {
+          "name": "price: models::Price",
+          "ordinal": 4,
+          "type_info": "Text"
+        }
+      ],
+      "parameters": {
+        "Right": 1
+      },
+      "nullable": [
+        true,
+        false,
+        false,
+        false,
+        false
+      ]
     }
   },
   "8556fa5ffa07322e35a16add4e40fa35bd543f7a1cb6d81b046ded4ec7bbb0be": {
@@ -534,6 +594,42 @@
         "Right": 2
       },
       "nullable": []
+    }
+  },
+  "aedd751cc7dcf48f77e8b00fba501ca65e0020dac15e6ba985bd61166c137531": {
+    "query": "\n        SELECT\n            closed_commit_txs.txid as \"commit_txid!: models::Txid\",\n            closed_refund_txs.txid as \"txid: models::Txid\",\n            closed_refund_txs.vout as \"vout: models::Vout\",\n            closed_refund_txs.payout as \"payout: models::Payout\"\n        FROM\n            closed_refund_txs\n        JOIN\n            closed_commit_txs on closed_commit_txs.cfd_id = closed_refund_txs.cfd_id\n        JOIN\n            closed_cfds on closed_cfds.id = closed_refund_txs.cfd_id\n        WHERE\n            closed_cfds.uuid = $1\n        ",
+    "describe": {
+      "columns": [
+        {
+          "name": "commit_txid!: models::Txid",
+          "ordinal": 0,
+          "type_info": "Text"
+        },
+        {
+          "name": "txid: models::Txid",
+          "ordinal": 1,
+          "type_info": "Text"
+        },
+        {
+          "name": "vout: models::Vout",
+          "ordinal": 2,
+          "type_info": "Int64"
+        },
+        {
+          "name": "payout: models::Payout",
+          "ordinal": 3,
+          "type_info": "Int64"
+        }
+      ],
+      "parameters": {
+        "Right": 1
+      },
+      "nullable": [
+        true,
+        false,
+        false,
+        false
+      ]
     }
   },
   "b15920dbf09664b23c4bf64bf552b865a3280e186d3bd9ac424dc2e46493b6ea": {
@@ -852,48 +948,6 @@
       "nullable": []
     }
   },
-  "ce102b72700c4c0e1742ff185706a52d8a4dca4e5eb0cde65affd8d5d01264eb": {
-    "query": "\n        SELECT\n            closed_commit_txs.txid as \"commit_txid: models::Txid\",\n            closed_cets.txid as \"txid: models::Txid\",\n            closed_cets.vout as \"vout: models::Vout\",\n            closed_cets.payout as \"payout: models::Payout\",\n            closed_cets.price as \"price: models::Price\"\n        FROM\n            closed_cets\n        JOIN\n            closed_commit_txs on closed_commit_txs.cfd_id = closed_cets.cfd_id\n        JOIN\n            closed_cfds on closed_cfds.id = closed_cets.cfd_id\n        WHERE\n            closed_cfds.uuid = $1\n        ",
-    "describe": {
-      "columns": [
-        {
-          "name": "commit_txid: models::Txid",
-          "ordinal": 0,
-          "type_info": "Text"
-        },
-        {
-          "name": "txid: models::Txid",
-          "ordinal": 1,
-          "type_info": "Text"
-        },
-        {
-          "name": "vout: models::Vout",
-          "ordinal": 2,
-          "type_info": "Int64"
-        },
-        {
-          "name": "payout: models::Payout",
-          "ordinal": 3,
-          "type_info": "Int64"
-        },
-        {
-          "name": "price: models::Price",
-          "ordinal": 4,
-          "type_info": "Text"
-        }
-      ],
-      "parameters": {
-        "Right": 1
-      },
-      "nullable": [
-        false,
-        false,
-        false,
-        false,
-        false
-      ]
-    }
-  },
   "d87c695f2f1f67e9acbc2ed4dac9a083738e82c52e419f5f025f8c4e327b4858": {
     "query": "\n            INSERT OR IGNORE INTO time_to_first_position\n            (\n                taker_id,\n                first_seen_timestamp\n            )\n            VALUES ($1, $2)\n            ",
     "describe": {
@@ -937,42 +991,6 @@
           "name": "price: models::Price",
           "ordinal": 3,
           "type_info": "Text"
-        }
-      ],
-      "parameters": {
-        "Right": 1
-      },
-      "nullable": [
-        false,
-        false,
-        false,
-        false
-      ]
-    }
-  },
-  "e81424a68a41a8d7f3ab624716aea93782b9a285547687e6f24cb393503fbaf3": {
-    "query": "\n        SELECT\n            closed_commit_txs.txid as \"commit_txid: models::Txid\",\n            closed_refund_txs.txid as \"txid: models::Txid\",\n            closed_refund_txs.vout as \"vout: models::Vout\",\n            closed_refund_txs.payout as \"payout: models::Payout\"\n        FROM\n            closed_refund_txs\n        JOIN\n            closed_commit_txs on closed_commit_txs.cfd_id = closed_refund_txs.cfd_id\n        JOIN\n            closed_cfds on closed_cfds.id = closed_refund_txs.cfd_id\n        WHERE\n            closed_cfds.uuid = $1\n        ",
-    "describe": {
-      "columns": [
-        {
-          "name": "commit_txid: models::Txid",
-          "ordinal": 0,
-          "type_info": "Text"
-        },
-        {
-          "name": "txid: models::Txid",
-          "ordinal": 1,
-          "type_info": "Text"
-        },
-        {
-          "name": "vout: models::Vout",
-          "ordinal": 2,
-          "type_info": "Int64"
-        },
-        {
-          "name": "payout: models::Payout",
-          "ordinal": 3,
-          "type_info": "Int64"
         }
       ],
       "parameters": {
@@ -1042,24 +1060,6 @@
         false,
         false,
         false,
-        false
-      ]
-    }
-  },
-  "eaa1ac5e11d036f67d5a6fd1b7c150dba3f59021faee1e89abb0f2cda4ca6af9": {
-    "query": "\n        SELECT\n            event_log_failed.created_at\n        FROM\n            event_log_failed\n        JOIN\n            failed_cfds on failed_cfds.id = event_log_failed.cfd_id\n        WHERE\n            failed_cfds.uuid = $1\n        ORDER BY event_log_failed.created_at ASC\n        LIMIT 1\n        ",
-    "describe": {
-      "columns": [
-        {
-          "name": "created_at",
-          "ordinal": 0,
-          "type_info": "Int64"
-        }
-      ],
-      "parameters": {
-        "Right": 1
-      },
-      "nullable": [
         false
       ]
     }

--- a/sqlite-db/src/closed.rs
+++ b/sqlite-db/src/closed.rs
@@ -855,7 +855,7 @@ async fn load_cet_settlement(
         models::Settlement::Cet,
         r#"
         SELECT
-            closed_commit_txs.txid as "commit_txid: models::Txid",
+            closed_commit_txs.txid as "commit_txid!: models::Txid",
             closed_cets.txid as "txid: models::Txid",
             closed_cets.vout as "vout: models::Vout",
             closed_cets.payout as "payout: models::Payout",
@@ -887,7 +887,7 @@ async fn load_refund_settlement(
         models::Settlement::Refund,
         r#"
         SELECT
-            closed_commit_txs.txid as "commit_txid: models::Txid",
+            closed_commit_txs.txid as "commit_txid!: models::Txid",
             closed_refund_txs.txid as "txid: models::Txid",
             closed_refund_txs.vout as "vout: models::Vout",
             closed_refund_txs.payout as "payout: models::Payout"
@@ -958,7 +958,7 @@ async fn load_creation_timestamp(
     let row = sqlx::query!(
         r#"
         SELECT
-            event_log.created_at
+            event_log.created_at as "created_at!: i64"
         FROM
             event_log
         JOIN

--- a/sqlite-db/src/failed.rs
+++ b/sqlite-db/src/failed.rs
@@ -296,7 +296,7 @@ async fn load_creation_timestamp(
     let row = sqlx::query!(
         r#"
         SELECT
-            event_log_failed.created_at
+            event_log_failed.created_at as "created_at!: i64"
         FROM
             event_log_failed
         JOIN

